From 31f5277e7215760c50fa77b5c1e188b6b3fca2cc Mon Sep 17 00:00:00 2001
From: Steve Traylen <steve.traylen@cern.ch>
Date: Sat, 23 May 2026 12:05:10 +0200
Subject: [PATCH] tests: fix test_uses_pprint for Python 3.15 pprint dict

Python 3.15 changed pprint's output for dicts with non-string keys

* https://docs.python.org/3.15/whatsnew/3.15.html#pprint
* https://github.com/python/cpython/issues/112632
---
 jsonschema/tests/test_exceptions.py | 71 +++++++++++++++++++++++++++++
 1 file changed, 71 insertions(+)

diff --git a/jsonschema/tests/test_exceptions.py b/jsonschema/tests/test_exceptions.py
index 358b9242..7863aaa6 100644
--- a/jsonschema/tests/test_exceptions.py
+++ b/jsonschema/tests/test_exceptions.py
@@ -1,5 +1,7 @@
 from unittest import TestCase
+import sys
 import textwrap
+import unittest
 
 import jsonpath_ng
 
@@ -593,6 +595,10 @@ def test_multiple_item_paths(self):
             schema_path=["items", 0, 1],
         )
 
+    @unittest.skipIf(
+        sys.version_info >= (3, 15),
+        "pprint dict format changed in 3.15, see test_uses_pprint_py315",
+    )
     def test_uses_pprint(self):
         self.assertShows(
             """
@@ -650,6 +656,71 @@ def test_uses_pprint(self):
             validator="maxLength",
         )
 
+    @unittest.skipIf(
+        sys.version_info < (3, 15),
+        "pprint dict format pre-3.15, see test_uses_pprint",
+    )
+    def test_uses_pprint_py315(self):
+        self.assertShows(
+            """
+            Failed validating 'maxLength' in schema:
+                {
+                    0: 0,
+                    1: 1,
+                    2: 2,
+                    3: 3,
+                    4: 4,
+                    5: 5,
+                    6: 6,
+                    7: 7,
+                    8: 8,
+                    9: 9,
+                    10: 10,
+                    11: 11,
+                    12: 12,
+                    13: 13,
+                    14: 14,
+                    15: 15,
+                    16: 16,
+                    17: 17,
+                    18: 18,
+                    19: 19,
+                }
+
+            On instance:
+                [
+                    0,
+                    1,
+                    2,
+                    3,
+                    4,
+                    5,
+                    6,
+                    7,
+                    8,
+                    9,
+                    10,
+                    11,
+                    12,
+                    13,
+                    14,
+                    15,
+                    16,
+                    17,
+                    18,
+                    19,
+                    20,
+                    21,
+                    22,
+                    23,
+                    24,
+                ]
+            """,
+            instance=list(range(25)),
+            schema=dict(zip(range(20), range(20))),
+            validator="maxLength",
+        )
+
     def test_does_not_reorder_dicts(self):
         self.assertShows(
             """
